internal/credential/credential.go

```diff
@@ -98,6 +98,11 @@ type Request struct {
 	Purpose  Purpose
 }
 
+type SshCertificate interface {
+	SshPrivateKey
+	Certifiacte() []byte
+}
```

Comment:
@tmessi is this necessary or can it be removed? I don't see any references in my current branch

Reply:
This is needed eventually, but for this PR it probably isn't necessary. Although the new credential that is returned, `sshCertCred`, does need to implement this interface.

Reply:
I'll leave in the interface (fixing the typo, of course) and then check in the tests that it matches